frontend/src/toolbar/elements/heatmapLogic.ts

```diff
@@ -137,12 +137,19 @@ export const heatmapLogic = kea<heatmapLogicType>([
         ],
     })),
 
-    selectors({
+    selectors(({ cache }) => ({
         elements: [
-            (selectors) => [selectors.elementStats, toolbarLogic.selectors.dataAttributes],
-            (elementStats, dataAttributes) => {
-                // cache all elements in shadow roots
-                const allElements = collectAllElementsDeep('*', document)
+            (selectors) => [
+                selectors.elementStats,
+                toolbarLogic.selectors.dataAttributes,
+                currentPageLogic.selectors.href,
+            ],
+            (elementStats, dataAttributes, href) => {
+                cache.pageElements = cache.lastHref == href ? cache.pageElements : collectAllElementsDeep('*', document)
+                cache.selectorToElements = cache.lastHref == href ? cache.selectorToElements : {}
```

Comment:
`cache.selectorToElements` !== `cache.selectorToElement` 🤔

Reply:
Ah, you saw my deliberate mistake intended to check if you were paying attention

![homer-hedge](https://user-images.githubusercontent.com/984817/214074597-0518c970-9d92-4c39-994a-61aa75de71cc.gif)